### ISSUE TITLE
Skybox can be seen in underground caves when the fog distance is very short

### ISSUE THREAD

**Issue by Lolothepro** [jellysquid3/Sodium#2634]
### Bug Description

In the caves, if a mod changes the fog start, the player can see the stars.

![image](https://github.com/user-attachments/assets/2473c009-76d3-4a31-b9a6-2e817dea8ed1)
Fog config: [config.json](https://github.com/user-attachments/files/16488571/config.json)

### Reproduction Steps

Install a mod that changes the Fog distance (like [Fog](https://modrinth.com/mod/fog))
Change the mod settings
![2024-08-04_18 17 52](https://github.com/user-attachments/assets/c2f8d066-e34a-44fc-aee8-c20af34377db)

### Log File

https://mclo.gs/UX1gjLf

### Crash Report

*

**Comment by jellysquid3** [comment]
I don't think this is easily fixed. The terrain shader is responsible for applying fog, and if there's no blocks between you and the sky (even if you are underground), the shader is not going to fill in that part of the sky.

Sodium likely makes this problem worse because it also culls geometry behind the fog. Can you see if disabling `Video Settings > Performance > Use Fog Occlusion` improves the issue?

**Comment by Lolothepro** [comment]
> I don't think this is easily fixed. The terrain shader is responsible for applying fog, and if there's no blocks between you and the sky (even if you are underground), the shader is not going to fill in that part of the sky.
> 
> Sodium likely makes this problem worse because it also culls geometry behind the fog. Can you see if disabling `Video Settings > Performance > Use Fog Occlusion` improves the issue?

Yeap
